libkv.go

```diff
@@ -1,3 +1,64 @@
+// Package libkv provides a Go native library to store metadata.
+//
+// The goal of libkv is to abstract common store operations for multiple
+// Key/Value backends and offer the same experience no matter which one of the
+// backend you want to use.
+//
+// For example, you can use it to store your metadata or for service discovery to
+// register machines and endpoints inside your cluster.
```

Comment:
I think these example usages are very specific to the way libkv is used by Docker. I think more generic examples would be good here.

(I realise this information was copied from the readme)